src/env_model_repository.cpp

```diff
@@ -75,6 +89,14 @@ void Item::addChild(std::shared_ptr<Item> child)
   children_.push_back(child);
   child->setParent(shared_from_this());
 }
+void Item::removeChild(std::shared_ptr<Item> child)
```

Comment:
its a good thing that you added a separate method, makes it more clear, but instead of passing a shared ptr as an argument (`std::shared_ptr<Item> child`) please pass `const std::string name`, because you are again making an unnecessary copy of the shared ptr you are trying to remove

Reply:
Ok, got it